### PR TITLE
[PhpUnitBridge] clarify silencing deprecations in tests

### LINKED ISSUE #7312: Clarify deprecation handling in the PhpUnitBridge
In https://symfony.com/doc/current/components/phpunit_bridge.html#mark-tests-as-legacy we say that there are four ways to declare a test as legacy:

> There are four ways to mark a test as legacy:
> - (Recommended) Add the @group legacy annotation to its class or method;
> - Make its class name start with the Legacy prefix;
> - Make its method name start with testLegacy instead of test;
> - Make its data provider start with provideLegacy or getLegacy.

However, this is not true. Prefix the name of a data provider method with `provideLegacy` or `getLegacy` does not influence the error handling of the test method, but is only useful if your data provider actually triggers deprecations.

### PR DESCRIPTION
**Pull request by xabbuh**
This resolves #7312.

### CODE REVIEW

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -73,15 +73,18 @@ in the **Unsilenced** section of the deprecation report.
...
 
 * Make its class name start with the ``Legacy`` prefix;
 
-* Make its method name start with ``testLegacy*()`` instead of ``test*()``;
+* Make its method name start with ``testLegacy*()`` instead of ``test*()``.
 
-* Make its data provider start with ``provideLegacy*()`` or ``getLegacy*()``.
+.. note::
+
+    You can make your data provider start with ``provideLegacy*()`` or ``getLegacy*()``
```
Comment (javiereguiluz) [inline components/phpunit_bridge.rst]: I'd like to have more information about this:

* If I name a data provider as `provideLegacy()` and `getLegacy()` ... will the tests that use that provider be considered legacy automatically?
* If I declare a test as legacy ... am I required to call its data provider as `provideLegacy()` and `getLegacy()` too?

Comment (javiereguiluz) [inline components/phpunit_bridge.rst]: So, any comment about my questions? Thanks!

**Comment by xabbuh** [comment]
@javiereguiluz Sorry for the delay. I pushed an update here. Do you think it's more clear now?

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -73,15 +73,25 @@ in the **Unsilenced** section of the deprecation report.
...
-* Make its data provider start with ``provideLegacy*()`` or ``getLegacy*()``.
+.. note::
+
+    If your data provider calls code that would usually trigger a deprecation,
+    you can prefix its name with ``provideLegacy`` or ``getLegacy`` to silent
+    these deprecations. If your data provider does not execute deprecated
+    code, it is not required to choose a special naming just because the
+    test being fed by the data provider is marked as legacy.
+
+    Also be aware that choosing one of the two legacy prefixes will not mark
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: "one of the" => "one of these"?

**Comment by javiereguiluz** [comment]
Thank you Christian.

**Pull request merged**